File: backend/src/main/java/com/project2/domain/member/dto/MemberProfileResponseDTO.java
package com.project2.domain.member.dto;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project2.domain.member.entity.Member;

import lombok.Getter;
import lombok.NonNull;

@Getter
public class MemberProfileResponseDTO {
	private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy년 MM월");

	private String nickname;
	@JsonProperty(required = true)
	@NonNull
	private String profileImageUrl;
	private long totalPostCount;
	private long totalFlowerCount;
	private long totalFlowingCount;
	private String createdMonthYear;
	private boolean isMe;

	public MemberProfileResponseDTO(Member member, long totalPostCount, long totalFlowerCount, long totalFlowingCount,
		boolean isMe) {
		this.nickname = member.getNickname();
		this.profileImageUrl = member.getProfileImageUrlOrDefaultUrl();
		this.totalPostCount = totalPostCount;
		this.totalFlowerCount = totalFlowerCount;
		this.totalFlowingCount = totalFlowingCount;
		this.createdMonthYear = formatCreatedDate(member.getCreatedDate());
		this.isMe = isMe;
	}

	private String formatCreatedDate(LocalDateTime createdDate) {
		return createdDate.format(DATE_FORMATTER);
	}
}

File: backend/src/main/kotlin/com/project2/domain/member/dto/MemberProfileResponseDTO.kt
package com.project2.domain.member.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.project2.domain.member.entity.Member
import java.time.format.DateTimeFormatter

data class MemberProfileResponseDTO(
        val nickname: String,
        @JsonProperty(required = true)
        val profileImageUrl: String,
        val totalPostCount: Long,
        val totalFlowerCount: Long,
        val totalFlowingCount: Long,
        val createdMonthYear: String,
        val isMe: Boolean
) {
    companion object {
        private val DATE_FORMATTER: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy년 MM월")

        fun from(
            member: Member,
            totalPostCount: Long,
            totalFlowerCount: Long,
            totalFlowingCount: Long,
            isMe: Boolean
        ): MemberProfileResponseDTO {
            return MemberProfileResponseDTO(
                    nickname = member.nickname,
                    profileImageUrl = member.getProfileImageUrlOrDefaultUrl(),
                    totalPostCount = totalPostCount,
                    totalFlowerCount = totalFlowerCount,
                    totalFlowingCount = totalFlowingCount,
                    createdMonthYear = member.createdDate.format(DATE_FORMATTER),
                    isMe = isMe
            )
        }
    }
}